app/src/app.tsx

```diff
@@ -28,6 +28,9 @@ const App = () => {
 
     useEffect(() => {
         onAuthStateChanged(setUser);
+    }, [setLoggedIn, setUserProfile]);
```

Comment:
Are these the right methods to include in this array?
I don't think it is quite right, see [here](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects).

Reply:
Ah, this is an indirect dependency.